code/Model/SiteTree.php

```diff
@@ -551,7 +560,7 @@ public static function page_type_classes()
             unset($classes[$baseClassIndex]);
         }
 
-        $kill_ancestors = [];
+        $kill_ancestors = self::config()->get('hide_pagetypes', Config::UNINHERITED) ?? [];
```

Comment:
Should we add an extension point if we're touching this? We had a few projects where it would come handy to be able to do this dynamically, e.g. for subsites, different user groups etc.

Reply:
I'd be happy for that to be implemented as a separate PR, but it's out of scope for this one. My intention here is to explicitly make the configuration property easier to work with since `hide_ancestor` is basically being abused at the moment for hiding arbitrary non-ancestor classes.

Reply:
In fact there's a whole separate card for improving the way that hidden pages is defined which includes scope for extensions: https://github.com/silverstripe/silverstripe-cms/issues/2207